front/src/applications/editor/tools/rangeEdition/speedSection/SpeedSectionEditionLayers.tsx

```diff
@@ -72,17 +75,29 @@ export const SpeedSectionEditionLayers = () => {
     return res;
   }, [interactionState, hoveredItem, entity, selectedSwitches]);
 
-  const highlightedTracksFeature: FeatureCollection = useMemo(() => {
+  const makeTrackRangeFeatures =
+    (flatEntity: EditorEntity) => (range: ApplicableTrackRange, i: number) => {
+      const trackState = trackSectionsCache[range.track];
+      return trackState?.type === 'success'
+        ? getTrackRangeFeatures(trackState.track, range, i, flatEntity.properties)
+        : [];
+    };
+
+  // dashed lines - - - (selected routes tracks in the case of speed restrictions)
+  const selectedTracksFeature: FeatureCollection = useMemo(() => {
     const flatEntity = flattenEntity(entity);
     // generate trackRangeFeatures
-    const trackRangeFeatures = Object.values(pick(routeElements, highlightedRoutes))
-      .flatMap((el) => el.trackRanges)
-      .flatMap((range, i) => {
-        const trackState = trackSectionsCache[range.track];
-        return trackState?.type === 'success'
-          ? getTrackRangeFeatures(trackState.track, range, i, flatEntity.properties)
-          : [];
-      }) as Feature<LineString | Point>[];
+    let trackRangeFeatures;
+    if (isSpeedRestriction) {
+      trackRangeFeatures = Object.values(pick(routeElements, highlightedRoutes))
+        .flatMap((el) => el.trackRanges)
+        .flatMap(makeTrackRangeFeatures(flatEntity)) as Feature<LineString | Point>[];
+    } else {
+      const trackRanges = entity.properties?.track_ranges || [];
+      trackRangeFeatures = trackRanges.flatMap(makeTrackRangeFeatures(flatEntity)) as Feature<
+        LineString | Point
+      >[];
+    }
```

Comment:
What do you think of that ?
```suggestion
    const trackRanges = isSpeedRestriction
      ? Object.values(pick(routeElements, highlightedRoutes)).flatMap((el) => el.trackRanges)
      : entity.properties?.track_ranges || [];
    const trackRangeFeatures = trackRanges.flatMap((range, i) =>
      extractTrackRangeFeatures(flatEntity, range, i)
    );
```

Reply:
I think the refacto is done with the function extractTrackRangeFeatures, we’re not repeating ourselves on the core stuff.

Reply:
I still prefer my suggestion, I think it's more readable. Can you change it ?

Reply:
done